libraries/authentication.rb

```diff
@@ -1,134 +1,107 @@
-# coding: UTF-8
+# frozen_string_literal: true
 
+#
 # Cookbook Name:: cerner_splunk
 # File Name:: authentication.rb
 
 require_relative 'databag'
 
 module CernerSplunk
-  ASSUMPTIONS =
-    {
-      'LDAP_strategies' => 'LDAP',
-      'cacheTiming' => 'Scripted',
-      'scriptPath' => 'Scripted',
-      'scriptSearchFilters' => 'Scripted',
-      'passwordHashAlgorithm' => 'Splunk'
-    }.freeze
-
   # Module contains functions to configure authentication in a Splunk system
   module Authentication
+    def self.determine_auth_type(settings)
+      config_mapping = {
+        'LDAP_strategies' => 'LDAP',
+        'cacheTiming' => 'Scripted',
+        'scriptPath' => 'Scripted',
+        'scriptSearchFilters' => 'Scripted',
+        'passwordHashAlgorithm' => 'Splunk'
+      }.freeze
+
+      probable_types = config_mapping.values_at(*settings.keys).uniq
```

Comment:
If settings['authType'] is set, you want to include this in the array that you then uniq over.